accepted/wasmtime-one-dot-oh.md

```diff
@@ -0,0 +1,558 @@
+binaries. Embeddings will all be tagged and released as appropriate to
+language-specific package managers (such as crates.io and PyPI). Wasmtime will
+always be released from the `main` branch of the Wasmtime repository itself,
+which means that all development is happening on `main` and once something lands
+it's guaranteed to be in the next release. Note that bug fixes and such for
+historical releases are discussed later in this proposal.
+
+Releasing a new version of Wasmtime every 4 weeks can be quite rapid for some
+users who don't necessarily want to stay up-to-date with the latest and greatest
+of Wasmtime, but still want the stability of a production-ready WebAssembly
+runtime. For these users, this leads well into the next section ...
+
+### Long-term Support Releases
+
+Wasmtime will support some releases for an extended period of time relative to
+other releases. These releases will be known as "long term support" releases or
+LTS releases. **Wasmtime will support two active LTS versions at any point in
+time, each supported for 10 months at a time**.
+
+For users who want to use Wasmtime but are not interested in upgrading monthly,
+this will be a suitable alternative where Wasmtime LTS versions will have the
+following properties while the version is considered "supported":
+
+* They will receive security fixes.
+* They will receive bug fixes related to correctness.
+* They will _not_ receive new features.
+
+For example if a new WebAssembly proposal is implemented it will not be
+backported to LTS versions. Additionally if a bug is fixed in a wasm proposal
+that is off-by-default in an LTS version it will also not be backported. Other
+bug fixes (and of course security fixes) will be backported to supported LTS
+versions.
+
+An LTS version is considered "supported" for 10 release cycles, or 40 weeks (~10
+months). Wasmtime will have two LTS versions at any point in time, with a new
+LTS happening every 5 releases (20 weeks, ~5 months). For ease of remembering
+what's an LTS and what isn't, all releases of Wasmtime divisible by 5 will be
+LTS version, with the exception of 1.0 being an LTS version as well. For example
+the LTS versions of Wasmtime will be 1.0, 5.0, 10.0, 15.0, etc.
+
+This cadence means that users who do not want to upgrade Wasmtime monthly will
+be expected to upgrade Wasmtime at least every 10 months, likely every 5 months.
+These upgrades are likely to be less "hassle-free" than each individual version
+upgrade since it will accumulate at least 5 releases worth of minor breaking
+changes. But in exchange, users of LTS versions have a 5-month window to
+upgrade while staying on a supported version.
+
+LTS versions will be maintained in separate branches of the Wasmtime repository.
+At this time it's expected that LTS versions will not needed separate branches
+in separate embedding repos and tags will suffice. The branch names for the
+Wasmtime repository will be `lts-latest` and `lts-oldest` for the most recent
```

Comment:
(a bit nit-picky, sorry) Instead of `lts-latest` / `lts-oldest`, how about `lts` / `lts-previous`? `lts-oldest` not technically the _oldest_ one, and previous doesn't leave room to any possible doubt imo.

Reply:
How about just many `lts-$VERSION` branches, the two most recent being the supported ones? No jumping `lts-latest` from 10.0 to 15.0 etc.

Reply:
Yeah in retrospect I don't think `latest` and `oldest` are good names, but I like @fitzgen's idea of just having a branch for all of them. It means they become irrelevant and ossified 10+ months after their creation but that's not really an issue with git

Reply:
whynotboth.jpg: Could have a moving tag for `current-lts` and `prev-lts` + fixed tags with version numbers. But I'll stop with the bikeshedding here, having tags with the version number would be already quite good too!